### ISSUE TITLE
Highlights in PDF render solid black in Evince/Zathura PDF readers

### ISSUE THREAD

**Issue by likyng** [koreader/koreader#5010]
* KOReader version: 2019.04 (22db71c07549ffc59196f1f27cd9bb162eacd6d7)
* Device: Kindle Paperweight 4

#### Issue
Highlights created in KOreader render solid black in a few other PDF viewers namely: Evince (3.28.4), Zathura (0.3.8, pdf-poppler 0.2.8) on Ubuntu 18.04. 

They render fine (yellow) on Adobe Acrobat DC and PDF X-Change Viewer on Windows 10.

I'm not well versed in the PDF specs, so it might very well be that the fault is entirely on Evince/Zathura for not implementing it completely/incorrectly. In this case, please close the issue...

#### Steps to reproduce
1. highlight a section in a PDF in KOreader
2. save the PDF
3. view the saved file in Evince/Zathura

**Comment by Frenzie** [comment]
This may have changed in newer MuPDF, but either way it would strongly surprise me if it weren't per-spec. ;-)

(Incidentally, I believe Zathura also has a MuPDF backend.)

**Comment by bradmont** [comment]
This also happens when opening the pdf in Okular (using koreader 2020.04.1), which is a shame since I use okular extensively in a research context. I gave Xournal a try and it also seems to have the same problem.

Just spitballing here, but maybe having an option to underline instead of highlighting would be a workaround?

**Comment by pazos** [comment]
Highlights made with KOReader are visible in both MuPDF and PDF.js (google chrome, polar bookshelf ...).

![2020-04-28_17-14](https://user-images.githubusercontent.com/975883/80504988-0a492980-8974-11ea-9553-e62ea87c5fbc.png)

They're also displayed on some mobile apps like MoonReader (android) or GoodReader (iOS). 

What seems impossible is to succesfully remove a highlight once it was saved to the file :/

**Comment by benoit-pierre** [comment]
With our updated MuPDF (1.24.2), testing with `test/paper.pdf`:

| Viewer | KOReader highlights shown? |
| --     |                       --   |
| Chromium (126.0.6478.114) | yes |
| Deepin-Reader (5.10.29) | yes |
| Evince (46.3) | no¹ |
| Firefox (128.0b7) | yes |
| Ghostscript (10.03.1) | yes |
| MuPDF (1.24.3) | yes |
| Okular (24.05.1) | no¹ |
| Xpdf (4.05) | no¹ |
| Xreader (4.2.1) | no¹ |
| Zathura (0.5.6) with zathura-pdf-mupdf (0.4.3) | yes |
| Zathura (0.5.6) with zathura-pdf-poppler (0.3.2) | no¹ |

- 1: highlights are not visible (text is rendered normally)

Basically: all poppler based readers don't show highlights.

**Comment by benoit-pierre** [comment]
But at least there don't render solid black anymore. ;)

**Comment by benoit-pierre** [comment]
Note: it's the same no matter the style (underline, strike-through, …).

**Comment by ironhak** [comment]
Will highlights ever be compatible between Okular and Korader?

**Comment by rossikos** [comment]
If anyone is still looking for a way to view KOReader highlights in poppler-based PDF viewers, I've found a workaround using the [Sioyek](https://github.com/ahrm/sioyek) PDF viewer. 

I made a [script](https://github.com/rossikos/koreader-to-sioyek) to import highights from KOReader to Sioyek. Once highlights have been imported, you can export the highlights to a poppler-compatible PDF using the :embed_annotations command.

That said, Sioyek also uses MuPDF, so if it is possible to embed poppler-compatible highlights with Sioyek, it should also be possible with KOReader, no?

**Comment by ironhak** [comment]
Are there any updates on this?
